File: contest/LeetCode_Weekly_Contest_204/Detect_Pattern_of_Length_M_Repeated_K_or_More_Times.cpp
class Solution {
public:
    bool containsPattern(vector<int>& arr, int m, int k) {
        int len = m * k;
        int n = arr.size();
        for (int i = 0; i + len <= n; ++i) {
            bool flag = true;
            for (int j = i; j < i+len; j += m) {
                for (int idx = 0; idx < m; idx++)
                    if (arr[i+idx] != arr[j+idx]) {
                        flag = false;
                        break;
                    }
                if (!flag) break;
            }
            if (flag) return true;
        }
        return false;
    }
};

File: contest/LeetCode_Weekly_Contest_204/Detect_Pattern_of_Length_M_Repeated_K_or_More_Times.py
class Solution(object):
    def containsPattern(self, arr, m, k):
        """
        :type arr: List[int]
        :type m: int
        :type k: int
        :rtype: bool
        """
        
        n, cnt = len(arr), m * k
        for i in xrange(0, n+1-cnt):
            flag = True
            for j in xrange(i, i+cnt, m):
                for k in xrange(0, m):
                    if arr[i+k] != arr[j+k]:
                        flag = False
                        break
                if not flag: break
            if flag: return True
        return False